src/Lookup/UnknownForeignRepositoryException.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Lookup;
+
+use Exception;
+use InvalidArgumentException;
+
+/**
+ * @since 3.7
+ *
+ * @license GPL-2.0+
+ */
+class UnknownForeignRepositoryException extends InvalidArgumentException {
+
+	/**
+	 * @var string
+	 */
+	private $repoName;
+
+	/**
+	 * @param string $repoName
+	 * @param string|null $message
+	 * @param Exception|null $previous
+	 */
+	public function __construct( $repoName, $message = null, Exception $previous = null ) {
+		$this->repoName = $repoName;
+
+		parent::__construct(
+			$message ?: 'Unknown repository name: ' . $repoName,
+			0,
+			$previous
+		);
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getRepoName() {
```

Comment:
Didn't we decided to call this `getRepositoryName` in other classes? This should be consistent, please.